test/atomicskiplist/atomic_skiplist_dscheck.ml

```diff
@@ -0,0 +1,28 @@
+(* This dscheck testcase is not terminating. *)
```

Comment:
This `dscheck` test takes forever (probably because of too many possible interleavings) but it can actually finish. 

It also displays a weird behavior (if `find` calls are replaced by `add` calls) which is due to randomness (as explained in [this `dscheck` issue](ocaml-multicore/dscheck#20)). It is easily avoided by adding  `Random.init 0` at the beginning of the test (right after `Atomic.trace (fun () ->`).